src/utils/hooks/use-redirect-location.js

```diff
@@ -144,3 +155,123 @@ export const useRedirectLocationInFlowchart = (
 
   return { errorMessage, invalidUrl };
 };
+
+export const useRedirectLocationInExperimentTracking = (data, reload) => {
+  const [enableComparisonView, setEnableComparisonView] = useState(false);
+  const [selectedRunIds, setSelectedRunIds] = useState([]);
+  const [activeTab, setActiveTab] = useState(tabLabels[0]);
+  const [errorMessage, setErrorMessage] = useState({});
+  const [invalidUrl, setInvalidUrl] = useState(false);
+
+  const { pathname, search } = useLocation();
+  const { toSelectedRunsPath } = useGeneratePathnameForExperimentTracking();
+
+  const matchedExperimentTrackingMainPage = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.experimentTracking.main],
+  });
+
+  const matchedSelectedView = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.experimentTracking.selectedView],
+  });
+
+  const matchedSelectedRuns = matchPath(pathname + search, {
+    exact: true,
+    path: [routes.experimentTracking.selectedRuns],
+  });
+
+  useEffect(() => {
+    if (
+      !matchedExperimentTrackingMainPage &&
+      !matchedSelectedRuns &&
+      !matchedSelectedView
+    ) {
+      setErrorMessage(errorMessages.experimentTracking);
+      setInvalidUrl(true);
+    }
+
+    if (matchedExperimentTrackingMainPage) {
+      if (data?.runsList.length > 0 && selectedRunIds.length === 0) {
+        setErrorMessage({});
+        setInvalidUrl(false);
+
+        /**
+         * If we return runs and don't yet have a selected run, set the first one
+         * as the default, with precedence given to runs that are bookmarked.
+         */
+        const bookmarkedRuns = data.runsList.filter((run) => {
+          return run.bookmark === true;
+        });
+
+        if (bookmarkedRuns.length > 0) {
+          const defaultRunFromBookmarked = bookmarkedRuns
+            .map((run) => run.id)
+            .slice(0, 1);
+          setSelectedRunIds(defaultRunFromBookmarked);
```

Comment:
Space between definitions and function calls.

```suggestion
          const defaultRunFromBookmarked = bookmarkedRuns
            .map((run) => run.id)
            .slice(0, 1);
            
          setSelectedRunIds(defaultRunFromBookmarked);
```